sqlx-core/src/postgres/types/lquery.rs

```diff
@@ -0,0 +1,303 @@
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::error::BoxDynError;
+use crate::postgres::{PgArgumentBuffer, PgTypeInfo, PgValueFormat, PgValueRef, Postgres};
+use crate::types::Type;
+use bitflags::bitflags;
+use std::fmt::{self, Display, Formatter};
+use std::io::Write;
+use std::ops::Deref;
+use std::str::FromStr;
+
+use crate::postgres::types::ltree::{PgLTreeLabel, PgLTreeParseError};
+
+/// Represents lquery specific errors
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+pub enum PgLQueryParseError {
+    #[error("lquery cannot be empty")]
+    EmptyString,
+    #[error("unexpected character in lquery")]
+    UnexpectedCharacter,
+    #[error("error parsing integer: {0}")]
+    ParseIntError(#[from] std::num::ParseIntError),
+    #[error("error parsing integer: {0}")]
+    LTreeParrseError(#[from] PgLTreeParseError),
+    /// LQuery version not supported
+    #[error("lquery version not supported")]
+    InvalidLqueryVersion,
+}
+
+/// Container for a Label Tree Query (`lquery`) in Postgres.
+///
+/// See https://www.postgresql.org/docs/current/ltree.html
+///
```

Comment:
Can you include the note I added to `PgLTree` about the Postgres version requirement? https://github.com/launchbadge/sqlx/blob/master/sqlx-core/src/postgres/types/ltree.rs#L30-L39

Reply:
Done